pyflagser/flagser_count.py

```diff
@@ -0,0 +1,127 @@
+"""Implementation of the python API for the cell count of the flagser C++
+library."""
+
+import numpy as np
+
+from ._utils import _extract_unweighted_graph, _extract_weighted_graph
+from .modules.flagser_count_pybind import compute_cell_count
+
+
+def flagser_count_unweighted(adjacency_matrix, min_dimension=0,
+                             max_dimension=np.inf, directed=True):
+    """Compute the cell count per dimension of a directed/undirected unweighted
+    flag complex.
+
+    From an adjacency_matrix construct all cells forming its associated flag
+    complex and compute their number per dimension.
+
+    Parameters
+    ----------
+    adjacency_matrix : 2d ndarray or scipy.sparse matrix of shape \
+        (n_vertices, n_vertices), required
+        Adjacency matrix of a directed/undirected unweighted graph. It is
+        understood as a boolean matrix. Off-diagonal, ``0`` or ``False`` values
+        denote abstent edges while non-``0`` or ``True`` values denote edges
+        which are present. Diagonal values are ignored.
+
+    directed : bool, optional, default: ``True``
+        If ``True``, computes homology for the directed flad complex determined
+        by `adjacency_matrix`. If ``False``, computes homology for the
+        undirected flag complex obtained by considering all edges as
+        undirected, and it is therefore sufficient (but not necessary)
+        to pass an upper-triangular matrix.
+
+    Returns
+    -------
+    out : list of int
+        Cell count (number of simplices) per dimension greater than or equal
+        than `min_dimension` and less than `max_dimension`.
+
+    Notes
+    -----
+    The input graphs cannot contain self-loops, i.e. edges that start and end
+    in the same vertex, therefore diagonal elements of the input adjacency
+    matrix will be ignored.
+
+    References
+    ----------
+    .. [1] D. Luetgehetmann, "Documentation of the C++ flagser library";
+           `GitHub: luetge/flagser <https://github.com/luetge/flagser/blob/\
+           master/docs/documentation_flagser.pdf>`_.
+
+    """
+    # Extract vertices and edges
+    vertices, edges = _extract_unweighted_graph(adjacency_matrix)
+
+    # Call flagser_count binding
+    cell_count = compute_cell_count(vertices, edges, directed)
+
+    return cell_count
+
+
+def flagser_count_weighted(adjacency_matrix, max_edge_length=None,
+                           min_dimension=0, max_dimension=np.inf,
+                           directed=True, filtration="max"):
+    """Compute the cell count per dimension of a directed/undirected
+    filtered flag complexes.
+
+    From an adjacency_matrix construct a filtered flag complex as a sequence of
+    its cells associated to their filtration values and compute the number of
+    cells per dimension.
```

Comment:
Is this the number of cells per dimension *at the end of the filtration*? If not, what is it?

Reply:
Yes, it is.

Reply:
Ok thanks! Should we say it explicitly? We could use the same wording as for Betti numbers in `flagser_weighted`.

Reply:
`max_edge_weight` is not defined yet So, I would leave there "at the end of the filtration.". In ``out``, I have updated the wordings.

Reply:
Thanks!